### PR TITLE
[MDL HL1] Fixed children nodes not deleted.

### PR DESCRIPTION
**Pull request by malortie**
As the HL1 MDL loader reads the file data, it creates and stores child nodes for the scene's root node.
At the end, it transfers every child contained in the list to the scene's root node. This avoids constantly resizing root node's **mChildren** array.

A problem occurs when an exception is thrown before node children from the list are added to the scene's root node. This results in a memory leak.

### CODE REVIEW

File: code/MDL/HalfLife/HL1MDLLoader.cpp [hunk 1]
```diff
@@ -125,12 +125,14 @@ void HL1MDLLoader::release_resources() {
         anim_headers_ = nullptr;
     }
 
-    if (rootnode_children_.size()) {
+    // Root has some children ndoes. so let's proceed them
```
Comment (malortie) [inline code/MDL/HalfLife/HL1MDLLoader.cpp]: **ndoes** should be "nodes".

Comment (kimkulling) [reply code/MDL/HalfLife/HL1MDLLoader.cpp]: Fixed!

File: code/MDL/HalfLife/HL1MDLLoader.cpp [hunk 2]
```diff
@@ -374,7 +374,7 @@ void HL1MDLLoader::read_texture(const Texture_HL1 *ptexture,
     for (outheight = 1; outheight < ptexture->height; outheight <<= 1)
         ;
 
-    if (outheight > BuffenLen) {
+    if (static_cast<size_t>(outheight) > BuffenLen) {
         outheight = BuffenLen;
```
Comment (malortie) [inline code/MDL/HalfLife/HL1MDLLoader.cpp]: There should probably be a **static_cast<int>**, otherwise this could raise a warning when building on x64.

**Comment by malortie** [comment]
@kimkulling 

There are a lot of failures when performing checks for code quality, as well as build errors. Is it possible to know why these errors were not raised in the earlier commits of HL1MDLLoader.cpp?

**Comment by malortie** [comment]
@kimkulling 

If you plan on making further changes in the method read_texture() in HL1MDLLoader.cpp, please review these PR before https://github.com/assimp/assimp/pull/2942  https://github.com/assimp/assimp/pull/2943. Thanks in advance.

**Comment by kimkulling** [comment]
Will do, thanks for the tipp!

**Comment by kimkulling** [comment]
Merged, thanks a lot for the contribution.

**Pull request merged**